### DISCUSSION TITLE
%ifdef problems after upgrade to 1.6

### DISCUSSION THREAD

**Discussion by jsuttinger** [beetbox/beets#4479] [Q&A] [answered]
Hello, 
beets 1.6
python 3.8
NOTE: After upgrading Beets I had some issues, corrected by upgrading 'confuse' to 2.0.

I didn't submit an issue as this is probably syntax on my end or something. 

Just upgraded to beets 1.6 the other day. Since, my %isdef in my path isn't being applied like it was in 1.5. 

`default: %the{$albumartist}/$albumartist - $original_year - $album %ifdef{label,($label) }[$format]/$track - $title`

Before I found %ifdef, if no label was found, the album folder would look like 'artist - year - album () [format]'. 

I added %ifdef like above and if no label was found beets would leave off (): 'artist - year - album [format]'

Since upgrading to 1.6 the '()' is back. I've attempted to mess with the syntax with no success.

**Comment by sampsyo** [comment, marked as answer]
Hi there! This is due to #3982, which fixed a disconnect between `%ifdef`'s intended behavior (and documentation) w/r/t its implementation.

For this use case—i.e., fixed fields that are present on every item—the right thing to use is `%if` rather than `%ifdef`. Try `%if{$label,($label) }`.

**Reply by jsuttinger** [reply]
That seems to do the trick, appreciate it.
